src/Typesense/ApiCall.ts

```diff
@@ -313,9 +313,11 @@ export default class ApiCall {
           }"`,
         );
         // this.logger.debug(error.stack)
-        this.logger.warn(
-          `Request #${requestNumber}: Sleeping for ${this.retryIntervalSeconds}s and then retrying request...`,
-        );
+        if (numTries < this.numRetriesPerRequest + 1) {
```

Comment:
If the code execution reaches here, then this condition will always be true, since this whole code block is inside a for loop. 

So I don't think this check is needed right?

Reply:
You're right except for one exception: the last try, therefore the condition will be met at each try but the last (at which point we don't want to log that we're trying again in Xs)

Reply:
Ah yes, you're right.